lib/vrl/stdlib/src/assert_eq.rs

```diff
@@ -0,0 +1,122 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct AssertEq;
+
+impl Function for AssertEq {
+    fn identifier(&self) -> &'static str {
+        "assert_eq"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "left",
+                kind: kind::ANY,
+                required: true,
+            },
+            Parameter {
+                keyword: "right",
+                kind: kind::ANY,
+                required: true,
+            },
+            Parameter {
+                keyword: "message",
+                kind: kind::BYTES,
+                required: false,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "success",
+                source: "asserteq!(true, true)",
+                result: Ok("true"),
+            },
+            Example {
+                title: "failure",
+                source: "asserteq!(true, false)",
+                result: Err(r#"function call error for "assert_eq" at (0:22): assertion failed"#),
```

Comment:
Looking below wouldn't this error message be: `"assertion failed: {} == {}"`

Reply:
Oops, yes, I neglected to update these examples. Thanks!

Reply:
NP, my first comment was asking about if we could message out the inputs 😄

Reply:
Whoops, didn't even catch the incorrect function names :|